library/src/main/java/com/novoda/downloadmanager/NotificationConfig.java

```diff
@@ -0,0 +1,53 @@
+package com.novoda.downloadmanager;
+
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.content.Context;
+import android.os.Build;
+import android.support.v4.app.NotificationCompat;
+
+public class NotificationConfig<T> implements NotificationChannelCreator, NotificationCreator<T> {
+
+    private final Context applicationContext;
+    private final String channelId;
+    private final String userFacingChannelDescription;
+    private final NotificationCustomiser<T> notificationCustomiser;
+    private final int importance;
+
+    public NotificationConfig(Context context, String channelId, String userFacingChannelDescription, NotificationCustomiser<T> customiser, int importance) {
+        this.applicationContext = context.getApplicationContext();
+        this.channelId = channelId;
+        this.userFacingChannelDescription = userFacingChannelDescription;
+        this.notificationCustomiser = customiser;
+        this.importance = importance;
+    }
+
+    @Override
+    public NotificationInformation createNotification(final T notificationPayload) {
+        return new NotificationInformation() {
+            @Override
+            public int getId() {
+                return notificationPayload.hashCode();
+            }
+
+            @Override
+            public Notification getNotification() {
+                NotificationCompat.Builder builder = new NotificationCompat.Builder(applicationContext, channelId);
+                return notificationCustomiser.customNotificationFrom(builder, notificationPayload);
```

Comment:
The `notificationCustomiser` field is passed in by the client. They will need to use the supplied `builder` in that implementation as it will have been pre-configured with the correct channel name. We can write Javadoc explaining that when we come to that task.